### DISCUSSION TITLE
alist uses casdoor for single sign-on, after configuring according to the official website manual, click to log in 403

### DISCUSSION THREAD

**Discussion by sanduo-656** [AlistGo/alist#5164] [Q&A]
alist uses casdoor for single sign-on, after configuring according to the official website manual, click to log in 403.
The authentication source is a self-built account of casdoor, and the application and callback address are written in accordance with regulations. The current test found that as long as the casdoor redirection address is changed to ip+port, you can log in normally.
But if you change it to an https domain name, a 403 will appear, and there will be no login redirection.
We have also tried changing the authentication source, such as DingTalk and Feishu, and this problem occurred.
![image](https://github.com/alist-org/alist/assets/40496018/93800d84-e8f7-4efc-add5-efa79a057779)
![image](https://github.com/alist-org/alist/assets/40496018/2fd22274-7d0b-4cc2-a81f-b93002c22bea)
GET https://pan.sand.com/api/auth/sso_callback?method=sso_get_token&code=1027be7b43c3b23c9713&state=http://10.255.254.195:8000

**Comment by anwen-anyi** [comment]
我这里使用正常 Casdoor 账号登录或者使用绑定的第三方账号登录都可以，可能是你的配置不正确。例如 Nginx配置？Casdoor重定向？
- 我也使用的是 `https://xx.xxx.xx` 域名进行登录的 ,没有使用 `IP:port` 进行登录。

![image](https://github.com/alist-org/alist/assets/56105412/742b9e55-1330-452f-9e87-851844c3c815)


-----

（下面视频可能不能在线看，不知道为什么，麻烦下载一下看 每一个文件就500K）
### 正常 Casdoor 账号登录 
https://github.com/alist-org/alist/assets/56105412/fa083168-586d-4887-a6ad-5b923f623cd4

### 使用钉钉绑定登录
https://github.com/alist-org/alist/assets/56105412/967d5028-02a3-4ccb-8325-04b6230dfae8

**Comment by sanduo-656** [comment]
嗯，我再尝试尝试。目前的确在alist外面有一层nginx 反向代理层。查看日志发现只有一个403的错误。不知道如何下手继续向下查询了。

**Comment by julor** [comment]
set origin = "your casdoor url" in app.conf
